web/src/invite.ts

```diff
@@ -475,10 +442,46 @@ function open_invite_user_modal(e: JQuery.ClickEvent<Document, undefined>): void
         };
 
         $("#invite-user-form .setup-tips-container").html(render_invite_tips_banner(context));
+
+        const toggler = components.toggle({
+            html_class: "invite_users_option_tabs large allow-overflow",
+            child_wants_focus: true,
+            values: [
+                {label: $t({defaultMessage: "Send invite email"}), key: "invite-email-tab"},
+                {label: $t({defaultMessage: "Create invite link"}), key: "invite-link-tab"},
+            ],
+            callback(_name, key) {
+                switch (key) {
+                    case "invite-email-tab":
+                        $("#invitee_emails_container").show();
+                        break;
+                    case "invite-link-tab":
+                        $("#invitee_emails_container").hide();
+                        break;
+                }
+                toggle_invite_submit_button();
+                reset_error_messages();
+            },
+        });
+        const $container = $("#invite_users_option_tabs_container");
+        if (!settings_data.user_can_invite_users_by_email()) {
+            toggler.disable_tab("invite-email-tab");
+            toggler.goto("invite-link-tab");
+        }
+        if (!settings_data.user_can_create_multiuse_invite()) {
+            toggler.disable_tab("invite-link-tab");
+        }
+        const $elem = toggler.get();
+        $container.append($elem);
+        setTimeout(() => {
+            $(".invite_users_option_tabs .ind-tab.selected").trigger("focus");
```

Comment:
I guess this was added to allow changing tabs using arrow key just after the modal opens. But this does not work as expected some times. When both the tabs are enabled, and the email tab is selected initially, I press the right arrow key after the modal opens, but the tab does not change. It only changes when I once press left arrow key and then it works as expected. I am not sure how to fix this.

Reply:
Yes, I've also noted that and it happens because we removed `selected:0` from the options passed to `components.toggle` (which we were earlier passing). In its absence, no tab is selected.

Here is a snippet from `components.ts:133`
```
// We should arguably default opts.selected to 0.
    if (typeof opts.selected === "number") {
        select_tab(opts.selected);
    }
```

I can add a commit which selects the 0th tab in case `opts.selected` is non-numeric.

Reply:
@sahil839 I've added `selected:0` which fixes the glitch. 

I tried defaulting to the 0th tab if `selected` is not provided, but it breaks a few things, like `settings_panel_menu`, so I think this should be worked on separately.

Reply:
Yeah, we can do that separately.

Reply:
So what is the next step in this PR?

Reply:
The "Fixes ..." line in commit message should end with a period. Looks good otherwise. I will mark it as ready for review by Tim.

Reply:
Sounds good, maybe worth opening an issue for that as a follow-up. I also notice that hitting `tab` once starting from the tab-switcher being selected seems to do nothing; you need to hit it twice to get to the first input. Maybe worth some follow-up investigation.